src/hooks/useAutoFocusInput.ts

```diff
@@ -1,24 +1,29 @@
 import {useFocusEffect} from '@react-navigation/native';
 import {useCallback, useContext, useEffect, useRef, useState} from 'react';
-import {InteractionManager} from 'react-native';
+import {InteractionManager, TextInput} from 'react-native';
 import CONST from '@src/CONST';
 import * as Expensify from '@src/Expensify';
 
-export default function useAutoFocusInput() {
+type UseAutoFocusInput = {
+    inputCallbackRef: (ref: TextInput | null) => void;
+};
+
+export default function useAutoFocusInput(): UseAutoFocusInput {
     const [isInputInitialized, setIsInputInitialized] = useState(false);
     const [isScreenTransitionEnded, setIsScreenTransitionEnded] = useState(false);
 
+    // @ts-expect-error TODO: Remove this when Expensify.js is migrated.
     const {isSplashHidden} = useContext(Expensify.SplashScreenHiddenContext);
 
-    const inputRef = useRef(null);
-    const focusTimeoutRef = useRef(null);
+    const inputRef = useRef<TextInput | null>(null);
+    const focusTimeoutRef = useRef<NodeJS.Timeout | null>(null);
 
     useEffect(() => {
         if (!isScreenTransitionEnded || !isInputInitialized || !inputRef.current || !isSplashHidden) {
             return;
         }
         InteractionManager.runAfterInteractions(() => {
-            inputRef.current.focus();
+            inputRef.current?.focus();
```

Comment:
Is this necessary? Wouldn't the above early return protect against `inputRef.current` being falsy?

Reply:
It is necessary, when I remove it, there is an error saying it could be undefined. Do you think we should remove it from the if statement in that case?

Reply:
I think it's unnecessary to have both the early return and the optional chaining. I think we should do one or the other. From what I've seen, we favor the optional chaining, so I would remove the early returns, yeah.

The early returns are good for when it's good to avoid unnecessary processing like `InteractionManager.runAfterInteractions()`.

In this case, maybe the early return is good to keep. If there is no `inputRef.current`, then skipping the `runAfterInteractions()` sounds like a good thing to do.

Reply:
Sorry, I said to do two different things there. Let's keep the early return and the optional chaining (since that throws an error). No changes necessary.

Reply:
@tgolen The optional chaining is necessary because `inputRef.current?.focus();` is inside `InteractionManager.runAfterInteractions` which is asynchronous and when it executes the callback inside, we have no guarantee that `inputRef.current` is still defined so TS will complain.